builtin/game/item_entity.lua

```diff
@@ -38,31 +38,32 @@ core.register_entity(":__builtin:item", {
 	age = 0,
 
 	set_item = function(self, itemstring)
-		local stack = ItemStack(itemstring or self.itemstring)
+		itemstring = itemstring or self.itemstring
+		if itemstring == "" then
```

Comment:
`itemstring` can also be a ItemStack, like used in `try_merge_with` and  `on_punch` to prevent possible information loss for upcoming features.